pennylane/interfaces/tf.py

```diff
@@ -170,15 +160,14 @@ def grad(grad_output, **tfkwargs):
             grad_input = tf.matmul(grad_output_row, jacobian)
             grad_input = tf.reshape(grad_input, [-1])
 
-            if non_diff_indices:
-                # TensorFlow requires we return a gradient of size (num_variables,)
-                res = np.zeros([qnode.num_variables])
-                indices = np.fromiter(diff_indices, dtype=np.int64)
-                res[indices] = grad_input
-                grad_input = tf.constant(res, dtype=dtype)
-
             grad_input_unflattened = unflatten_tf(grad_input, input_)[0]
 
+            for idx, requires in enumerate(requires_grad):
+                # If a particular input argument is non-differentiable,
+                # replace the corresponding position in the gradient with None.
+                if not requires:
+                    grad_input_unflattened[idx] = None
+
```

Comment:
How come tf interface is the only one that needs to do this?

Reply:
This is specific to the TF custom gradient API. I assume PyTorch and Autograd do this bookkeeping themselves.